aten/src/ATen/native/sparse/cuda/SparseBlasImpl.cpp

```diff
@@ -832,21 +832,73 @@ void addmm_out_sparse_csr(
     const Scalar& beta,
     const Scalar& alpha,
     const Tensor& result) {
-  if (mat1.layout() == kSparseBsr && mat2.layout() == kStrided && result.layout() == kStrided) {
-    return block_sparse_mm(mat1, mat2, beta, alpha, result);
-  }
-  if (mat1.is_sparse_csr() && mat2.layout() == kStrided && result.layout() == kStrided) {
-    return spmm(mat1, mat2, beta, alpha, result);
-  }
-  if (mat1.layout() == kStrided && mat2.is_sparse_csr() && result.layout() == kStrided) {
-    // TODO: Add native CSC support via cuSPARSE if supported.
-    return spmm(mat2.transpose(0, 1).to_sparse_csr(), mat1.transpose(0, 1), beta, alpha, result.transpose(0, 1));
-  }
-  if (mat1.is_sparse_csr() && mat2.is_sparse_csr() && result.is_sparse_csr()) {
-    return spgemm(mat1, mat2, beta, alpha, result);
+  TORCH_INTERNAL_ASSERT(
+      !((mat1.layout() == kStrided) && (mat2.layout() == kStrided) &&
+        (result.layout() == kStrided)),
+      "Expected at least one sparse input");
+  if (mat1.layout() == kStrided) {
```

Comment:
I will add a comment to explain the structure here, but the nested structure makes it easier to parse out the supported combinations. 

Outer condition is mat1 layout, then mat2, then mat3. Conditions are ordered `strided, csr, csc, bsr, bsc` If a combination is not valid it is omitted. All unhandled combinations result in the `TORCH_CHECK(false ...` generating an informative error message. 

The same structure has been applied to the corresponding function for mkl.